rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindInstance.scala

```diff
@@ -26,15 +32,142 @@ object BitcoindInstance {
+
+    val config = ConfigFactory.parseFile(file)
+
+    // skip .resolve because Bitcoin Core only uses simple INI-like notation,
+    // not environment variable substitution
+    fromConfig(config)
+  }
+
+  def fromConfig(
+      config: Config,
+      datadir: File = Paths
+        .get(Properties.userHome, ".bitcoin")
+        .toFile): BitcoindInstance = {
+    val network = getNetwork(config)
+
+    val uri = getUri(config, network)
+    val rpcUri = getRpcUri(config, network)
+
+    val username = config.getString("rpcuser")
+    val password = config.getString("rpcpassword")
+    val authCredentials =
+      BitcoindAuthCredentials(username = username,
+                              password = password,
+                              rpcPort = rpcUri.getPort,
+                              datadir = datadir)
+
+    val zmqBlockUri = getZmqBlockUri(config)
+    val zmqTXUri = getZmqTxUri(config)
+
+    BitcoindInstance(network,
+                     uri,
+                     rpcUri,
+                     authCredentials,
+                     zmqBlockUri,
+                     zmqTXUri)
+  }
+
+  private def isSet(config: Config, path: String): Boolean = {
+    Try(config.getInt(path))
+      .map(_ == 1)
+      .getOrElse(false)
+  }
+
+  private def getNetwork(config: Config): BitcoinNetwork = {
+    val isTestnet = isSet(config, path = "testnet")
+    val isRegTest = isSet(config, path = "regtest")
+
+    (isRegTest, isTestnet) match {
+      case (true, true) =>
+        throw new IllegalArgumentException(
+          """"Cannot set both "regtest" and "testnet" options""")
+      case (true, false)  => RegTest
+      case (false, true)  => TestNet3
+      case (false, false) => MainNet
+    }
+  }
+
+  private def getUri(config: Config, network: NetworkParameters): URI = {
+    val port = Try(config.getInt("port")).getOrElse(network.port)
+    val host = Try(config.getString("bind")).getOrElse("localhost")
+    val uriT = Try {
+      new URI(s"http://$host:$port") // how to specify http/https?
+    }
+
+    uriT match {
+      case Success(uriSuccess) => uriSuccess
+      case Failure(exception) =>
+        throw new IllegalArgumentException(
+          s"Could not construct URI from host $host and port $port",
+          exception)
+    }
+
   }
+
+  private def getRpcUri(config: Config, network: NetworkParameters) = {
+    val rpcPort = Try(config.getInt("rpcport")).getOrElse(network.rpcPort)
+    val rpcHost = Try(config.getString("rpcbind")).getOrElse("localhost")
+
+    val rpcUriT = Try {
+      new URI(s"http://$rpcHost:$rpcPort")
+    }
+    rpcUriT match {
+      case Success(uriSuccess) => uriSuccess
+      case Failure(exception) =>
+        throw new IllegalArgumentException(
+          s"Could not construct URI from host $rpcHost and port $rpcPort",
+          exception)
+    }
+  }
+
+  private def getZmqUri(config: Config, path: String): Option[URI] = {
+    require(path == "zmqpubrawblock" || path == "zmqpubrawtx")
```

Comment:
Just an FYI, there is also `zmqpubhashblock` and `zmqpubhashtx` too i believe. Maybe we should just create a trait called `ZmqConfig`? 

```scala 
trait ZmqConfig { 
  def rawBlock  = ...
  def hashBlock = ...
  def hashTx = ...
  def rawTx = ...
}
```